dev/Interop/StoragePickers/FileTypeChoicesMap.h

```diff
@@ -28,7 +37,49 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
         winrt::Windows::Foundation::Collections::IIterator<winrt::Windows::Foundation::Collections::IKeyValuePair<hstring, winrt::Windows::Foundation::Collections::IVector<hstring>>> First() const;
 
     private:
-        winrt::Windows::Foundation::Collections::IMap<hstring, winrt::Windows::Foundation::Collections::IVector<hstring>> m_innerMap{
-            single_threaded_map<hstring, winrt::Windows::Foundation::Collections::IVector<hstring>>() };
+        // Shared ownership keeps insertion-ordered entries alive for any outstanding views/iterators.
+        std::shared_ptr<FileTypeChoiceVector> m_orderedMap;
+
+        // Helper methods
+        FileTypeChoiceVector::const_iterator FindKey(hstring const& key) const noexcept;
+        FileTypeChoiceVector::iterator FindKey(hstring const& key) noexcept;
+    };
+
+    // Custom iterator to maintain insertion order
+    struct OrderedMapIterator : implements<OrderedMapIterator,
+        winrt::Windows::Foundation::Collections::IIterator<winrt::Windows::Foundation::Collections::IKeyValuePair<hstring, winrt::Windows::Foundation::Collections::IVector<hstring>>>>
+    {
+        OrderedMapIterator(std::shared_ptr<FileTypeChoiceVector const> map);
```

Comment:
curious about the choice to copy/pass-by-value/move.  your earlier implementation was pass-by-const-ref/copy.  either way there's a copy, but the latter is simpler and you're passing one pointer (const ref) instead of two (std::shared_ptr value).